lib/spack/spack/solver/asp.py

```diff
@@ -516,7 +516,7 @@ def _compute_specs_from_answer_set(self):
                 node = SpecBuilder.make_node(pkg=providers[0])
             candidate = answer.get(node)
 
-            if candidate and candidate.satisfies(input_spec):
+            if candidate and candidate.build_spec.satisfies(input_spec):
```

Comment:
Shouldn't both the spec and the build spec satisfy the input spec?  Seems like this at least needs a note (a longer PR description might also help)